File: CS/WindowsApplication3/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.Controls;
using System.Drawing;
using DevExpress.XtraEditors.Repository;


namespace DXSample {
    public partial class Form1: Form {
        public Form1() {
            InitializeComponent();
        }
        public void InitData() {
            recordBindingSource.DataSource = DataHelper.GetData(100);
        }
       
        private void Form1_Load(object sender, EventArgs e) {
            this.InitData();
        }
       
        int rowHandle = -1;

        private void gridView1_MouseMove(object sender, MouseEventArgs e) {
            GridView view = sender as GridView;
            GridHitInfo hi = view.CalcHitInfo(e.Location);

            if(hi.InRowCell && hi.Column.FieldName == "Text") 
                rowHandle = hi.RowHandle;
            else
                rowHandle = -1;
           view.Invalidate(); 
        }

        private void gridView1_CustomDrawCell(object sender, DevExpress.XtraGrid.Views.Base.RowCellCustomDrawEventArgs e) {
            if(e.Column.FieldName == "Text" && e.RowHandle == rowHandle) {
                GridCellInfo cellInfo = e.Cell as GridCellInfo;
                ButtonEditViewInfo info = cellInfo.ViewInfo as ButtonEditViewInfo;
                Rectangle textRect = info.MaskBoxRect;
                textRect.Offset(e.Bounds.X, e.Bounds.Y);
                EditorButtonObjectInfoArgs args = new EditorButtonObjectInfoArgs(e.Cache, info.Item.Buttons[0], e.Appearance);
                Rectangle minBounds = info.EditorButtonPainter.CalcObjectMinBounds(args);
                args.Bounds = new Rectangle(e.Bounds.Right - minBounds.Width, e.Bounds.Y + (e.Bounds.Height - minBounds.Height) / 2, minBounds.Width, minBounds.Height);
                info.EditorButtonPainter.DrawObject(args);
                e.Cache.DrawString(e.DisplayText, e.Appearance.Font, e.Appearance.GetForeBrush(e.Cache), textRect, e.Appearance.GetStringFormat());
                e.Handled = true;
            }
        }

    }
}

File: VB/WindowsApplication3/Form1.vb
Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.Controls
Imports System.Drawing
Imports DevExpress.XtraEditors.Repository


Namespace DXSample
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub
        Public Sub InitData()
            recordBindingSource.DataSource = DataHelper.GetData(100)
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
            Me.InitData()
        End Sub

        Private rowHandle As Integer = -1

        Private Sub gridView1_MouseMove(ByVal sender As Object, ByVal e As MouseEventArgs) Handles gridView1.MouseMove
            Dim view As GridView = TryCast(sender, GridView)
            Dim hi As GridHitInfo = view.CalcHitInfo(e.Location)

            If hi.InRowCell AndAlso hi.Column.FieldName = "Text" Then
                rowHandle = hi.RowHandle
            Else
                rowHandle = -1
            End If
           view.Invalidate()
        End Sub

        Private Sub gridView1_CustomDrawCell(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Base.RowCellCustomDrawEventArgs) Handles gridView1.CustomDrawCell
            If e.Column.FieldName = "Text" AndAlso e.RowHandle = rowHandle Then
                Dim cellInfo As GridCellInfo = TryCast(e.Cell, GridCellInfo)
                Dim info As ButtonEditViewInfo = TryCast(cellInfo.ViewInfo, ButtonEditViewInfo)
                Dim textRect As Rectangle = info.MaskBoxRect
                textRect.Offset(e.Bounds.X, e.Bounds.Y)
                Dim args As New EditorButtonObjectInfoArgs(e.Cache, info.Item.Buttons(0), e.Appearance)
                Dim minBounds As Rectangle = info.EditorButtonPainter.CalcObjectMinBounds(args)
                args.Bounds = New Rectangle(e.Bounds.Right - minBounds.Width, e.Bounds.Y + (e.Bounds.Height - minBounds.Height) \ 2, minBounds.Width, minBounds.Height)
                info.EditorButtonPainter.DrawObject(args)
                e.Cache.DrawString(e.DisplayText, e.Appearance.Font, e.Appearance.GetForeBrush(e.Cache), textRect, e.Appearance.GetStringFormat())
                e.Handled = True
            End If
        End Sub

    End Class
End Namespace
